From c3b532a63f1d6c06df61f613e27d205333e953fd Mon Sep 17 00:00:00 2001
From: "Simental Magana, Marcos" <marcos.simental.magana@intel.com>
Date: Tue, 7 Apr 2015 11:25:36 -0500
Subject: [PATCH] Remove RC4 from crypto

---
 lib/crypto/c_src/crypto.c | 63 -----------------------------------------------
 1 file changed, 63 deletions(-)

diff --git a/lib/crypto/c_src/crypto.c b/lib/crypto/c_src/crypto.c
index 26e2486..7ded7ad 100644
--- a/lib/crypto/c_src/crypto.c
+++ b/lib/crypto/c_src/crypto.c
@@ -47,7 +47,6 @@
 #include <openssl/ripemd.h>
 #include <openssl/bn.h>
 #include <openssl/objects.h>
-#include <openssl/rc4.h>
 #include <openssl/rc2.h>
 #include <openssl/blowfish.h>
 #include <openssl/rand.h>
@@ -230,9 +229,6 @@ static ERL_NIF_TERM rsa_verify_nif(ErlNifEnv* env, int argc, const ERL_NIF_TERM
 static ERL_NIF_TERM aes_cbc_crypt(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[]);
 static ERL_NIF_TERM aes_ige_crypt_nif(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[]);
 static ERL_NIF_TERM do_exor(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[]);
-static ERL_NIF_TERM rc4_encrypt(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[]);
-static ERL_NIF_TERM rc4_set_key(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[]);
-static ERL_NIF_TERM rc4_encrypt_with_state(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[]);
 static ERL_NIF_TERM rc2_cbc_crypt(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[]);
 static ERL_NIF_TERM rsa_sign_nif(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[]);
 static ERL_NIF_TERM dss_sign_nif(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[]);
@@ -362,9 +358,6 @@ static ErlNifFunc nif_funcs[] = {
     {"aes_cbc_crypt", 4, aes_cbc_crypt},
     {"aes_ige_crypt_nif", 4, aes_ige_crypt_nif},
     {"do_exor", 2, do_exor},
-    {"rc4_encrypt", 2, rc4_encrypt},
-    {"rc4_set_key", 1, rc4_set_key},
-    {"rc4_encrypt_with_state", 2, rc4_encrypt_with_state},
     {"rc2_cbc_crypt", 4, rc2_cbc_crypt},
     {"rsa_sign_nif", 3, rsa_sign_nif},
     {"dss_sign_nif", 3, dss_sign_nif},
@@ -2292,62 +2285,6 @@ static ERL_NIF_TERM do_exor(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[])
     return ret;
 }
 
-static ERL_NIF_TERM rc4_encrypt(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[])
-{/* (Key, Data) */
-    ErlNifBinary key, data;
-    RC4_KEY rc4_key;
-    ERL_NIF_TERM ret;
-
-    CHECK_OSE_CRYPTO();
-
-    if (!enif_inspect_iolist_as_binary(env,argv[0], &key)
-	|| !enif_inspect_iolist_as_binary(env,argv[1], &data)) {
-	return enif_make_badarg(env);
-    }
-    RC4_set_key(&rc4_key, key.size, key.data);
-    RC4(&rc4_key, data.size, data.data,
-	enif_make_new_binary(env, data.size, &ret));
-    CONSUME_REDS(env,data);
-    return ret;
-}   
-
-static ERL_NIF_TERM rc4_set_key(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[])
-{/* (Key) */
-    ErlNifBinary key;
-    ERL_NIF_TERM ret;
-
-    CHECK_OSE_CRYPTO();
-
-    if (!enif_inspect_iolist_as_binary(env,argv[0], &key)) {
-	return enif_make_badarg(env);
-    }
-    RC4_set_key((RC4_KEY*)enif_make_new_binary(env, sizeof(RC4_KEY), &ret),
-		key.size, key.data);        
-    return ret;
-}
-
-static ERL_NIF_TERM rc4_encrypt_with_state(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[])
-{/* (State, Data) */
-
-    ErlNifBinary state, data;
-    RC4_KEY* rc4_key;
-    ERL_NIF_TERM new_state, new_data;
-
-    CHECK_OSE_CRYPTO();
-
-    if (!enif_inspect_iolist_as_binary(env,argv[0], &state)
-	|| state.size != sizeof(RC4_KEY)
-	|| !enif_inspect_iolist_as_binary(env,argv[1], &data)) {
-	return enif_make_badarg(env);
-    }
-    rc4_key = (RC4_KEY*)enif_make_new_binary(env, sizeof(RC4_KEY), &new_state);
-    memcpy(rc4_key, state.data, sizeof(RC4_KEY));
-    RC4(rc4_key, data.size, data.data,
-	enif_make_new_binary(env, data.size, &new_data));
-    CONSUME_REDS(env,data);
-    return enif_make_tuple2(env,new_state,new_data);
-}   
-
 static ERL_NIF_TERM rc2_cbc_crypt(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[])
 {/* (Key,IVec,Data,IsEncrypt) */
     ErlNifBinary key_bin, ivec_bin, data_bin;
-- 
1.9.3

